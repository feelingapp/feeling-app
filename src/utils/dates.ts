import { differenceInMonths, addMonths, addYears } from "date-fns/esm"

// Specify limits for the calendar in the Me tab
export const MIN_DATE = new Date(2020, 0, 1)
export const MAX_DATE = addYears(new Date(), 1)

export const getDateFromIndex = (index: number): Date =>
  addMonths(MIN_DATE, index)

export const getIndexFromDate = (date: Date): number =>
  differenceInMonths(date, MIN_DATE)
