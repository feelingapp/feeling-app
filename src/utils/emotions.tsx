import AngryIcon from "../assets/emojis/angry.svg"
import UpsetIcon from "../assets/emojis/upset.svg"
import UnsureIcon from "../assets/emojis/unsure.svg"
import GreatIcon from "../assets/emojis/great.svg"
import AmazingIcon from "../assets/emojis/amazing.svg"

export enum Emotion {
  ANGRY = "angry",
  UPSET = "upset",
  UNSURE = "unsure",
  GREAT = "great",
  AMAZING = "amazing",
}

export function getEmotionColor(emotion: Emotion): string {
  switch (emotion) {
    case Emotion.ANGRY:
      return "var(--purple-color)"
    case Emotion.UPSET:
      return "var(--blue-color)"
    case Emotion.UNSURE:
      return "var(--green-color)"
    case Emotion.GREAT:
      return "var(--yellow-color)"
    case Emotion.AMAZING:
      return "var(--red-color)"
    default:
      return "var(--light-grey-color)"
  }
}

export function renderEmoji(emotion: Emotion): JSX.Element {
  switch (emotion) {
    case Emotion.ANGRY:
      return <AngryIcon />
    case Emotion.UPSET:
      return <UpsetIcon />
    case Emotion.UNSURE:
      return <UnsureIcon />
    case Emotion.GREAT:
      return <GreatIcon />
    case Emotion.AMAZING:
      return <AmazingIcon />
    default:
      return undefined
  }
}
