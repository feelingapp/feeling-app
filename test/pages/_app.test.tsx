import { shallow } from "enzyme"
import MyApp from "../../src/pages/_app"
import ComponentStub from "../__mocks__/ComponentStub"
import * as Utils from "next/dist/next-server/lib/utils"

jest.mock("next/link", () => ({}))
jest.mock("next/router", () => ({}))
Utils.getDisplayName = jest.fn()

describe("App: ", () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it("matches snapshot when on landing page", () => {
    Utils.getDisplayName.mockImplementation(() => "Landing")

    const wrapper = shallow(<MyApp Component={ComponentStub}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it("matches snapshot when on App", () => {
    Utils.getDisplayName.mockImplementation(() => "App")

    const wrapper = shallow(<MyApp Component={ComponentStub}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
