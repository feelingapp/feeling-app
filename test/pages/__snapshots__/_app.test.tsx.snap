// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App:  matches snapshot when on App 1`] = `
<Fragment>
  <Main>
    <ComponentStub />
    <TabBar />
  </Main>
</Fragment>
`;

exports[`App:  matches snapshot when on landing page 1`] = `
<Fragment>
  <main>
    <ComponentStub />
  </main>
</Fragment>
`;
